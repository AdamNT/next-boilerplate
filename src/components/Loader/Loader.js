import styled, { keyframes } from 'styled-components';

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const Loader = styled.div`
  animation: ${rotate} 2s linear infinite;
  border-radius: 50%;
  border-top: 8px solid #3498db;
  border: 8px solid #f3f3f3;
  height: 40px;
  margin-left: auto;
  margin-right: auto;
  margin-top: 40px;
  width: 40px;
`;

export default Loader;
