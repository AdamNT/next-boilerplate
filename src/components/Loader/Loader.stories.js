// #region Global Imports
import React from 'react';
// #endregion Global Imports

// #region Local Imports
import Loader from './Loader';

// #endregion Local Imports

export default {
  title: 'Design System/Atoms/Loader',
  component: Loader,
};

export const Primary = () => <Loader />;
