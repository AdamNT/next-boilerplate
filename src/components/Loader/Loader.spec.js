import React from 'react';
import renderer from 'react-test-renderer';

import Loader from './Loader';

describe('Loader', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Loader />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
