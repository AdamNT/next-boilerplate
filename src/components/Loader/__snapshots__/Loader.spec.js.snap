// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader snapshot renders 1`] = `
<div
  className="Loader-art2jo-0 hqsEcO"
/>
`;
